Create arsalaan-alam.md

Arsalaan-Alam opened the pull request:
## Describe your project in 1-3 sentences

A drone that can deliver stuff within a range of 50 metres autonomously. It is capable of carrying about 50 grams of weight.

## Hack Club Slack handle

<!-- Join the Hack Club Slack at https://hackclub.com/slack - please write your handle ABOVE this line :)-->

## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

bellesea reviewed (changes requested):
@Arsalaan-Alam this is super cool! i have a few questions:
- how do you plan on overwriting the original firmware? what exactly would you be doing?
- do you need more than one TOF sensor so it can sense in more directions?
- how do you plan on powering your arduino?
- how would power your arduino nano?
- are you using an arduino nano or arduino due?
- is the size of the cube large enough to carry the items you need it to?

thank you!

Arsalaan-Alam commented:
@bellesea 
1. Not using an already prebuilt drone now, would be assembling my own, So I do not need to overwrite any code, I'll be writing my own from scratch. The drone consists of an APM controller. I would be downloading the ArduPilot software to programme the controller: https://ardupilot.org/planner/docs/mission-planner-installation.html

2. Not sure if 4 ToF sensors would be efficient enough, could make the drone clumsy. Thinking of having 2 ToF sensors at max, as I said the drone would be semi-automatic not fully automatic. I suppose it would require manual supervision at all times.

3,4,5. Would be powering my Arduino with a 5500 MAH LIPO battery. Could be overloaded, but I'd figure something out. Also have switched from Arduino nano to Arduino UNO as I've changed my drown. The new PCB can max lift around 1 kg of weight. I estimate the Arduino + sensors to be under 40 grams.

6. Updated the cube's dimensions to 10x10x10 CM. As said before, the drone can lift more weight now.

farhanalam25 commented:
Hi, I'm Farhan Alam, and I'm the little brother of Arsalaan. I'm currently in my freshman year in high school, and I'm down to collaborate with my brother on this project. Excited to build AirDrop!

[x] I am a current high school student (or younger).
[x] I have submitted an RSVP at https://hackclub.com/winter
[x] I am a member of the Hack Club Slack (Join at https://hackclub.com/slack)
[x] I have self-reviewed my project proposal.

bellesea reviewed (changes requested):
hey @Arsalaan-Alam thanks for pushing the changes!

a few questions:
- how do you plan on modifying the drone's original codebase? is the drone open-sourced? would it be better to build an open-sourced drone from scratch?
- if you're running the obstacle detection calculations on your arduino how would it communicate with your flight controller?
- are you using an arduino nano or uno? i see a reference to the nano above? and, like mentioned before, do you need more than one ToF sensor?

thank you :)

Arsalaan-Alam commented:
1. The drone is open sourced. As I said before, I would have to program it myself from scratch. The ardupilot software is available online, which I'd download to program the APM controller chip.
2. https://youtu.be/KI76nA7P5MI - Here, I'll be referencing this video to make the obstacle detection possible.
3. I'm using an Arduino UNO, and another update is that I would be using 6 ultrasonic sensors instead of the ToF sensor. 
4. The 6 ultrasonic sensors would give 360° information to the Arduino UNO. I could choose Arduino Nano too, and it might give better performance, but the budget doesn't allow it 
5. Please refer to the YouTube video to see how the obstacle detection programming would be done

bellesea commented on arsalaan-alam.md:
```diff
@@ -0,0 +1,52 @@
+---
+name: "@Arsalaan-Alam"
```
```suggestion
name: ["@Arsalaan-Alam", "@farhanalam25"]
```

bellesea reviewed (approved):
@Arsalaan-Alam and @farhanalam25 thank you so much for pushing changes and for elaborating on your plan :)) this is super awesome!

an awesome member of @hackclub/bank-ops will be reaching out soon with next steps :)

also, approving this to move forward but could you please swap out your ToF sensor for the 6 ultrasonic sensors + add the link you're referencing within your file? thank you!

hackliv reviewed (approved):
@Arsalaan-Alam this is so cool! Thanks for sharing more about your background making things fly too! 😃 I'm excited to see your project come to life! Dming you now on Slack

sarthakpandit0408 commented:
Hi, I'm Sarthak Pandit, a high school senior at DLF Public School Ghaziabad, I'm down to collaborate with Arsalaan and Farhan to build AirDrop. Excited to ship this project!

[x] I am a current high school student (or younger).
[x] I have submitted an RSVP at https://hackclub.com/winter
[x] I am a member of the Hack Club Slack (Join at https://hackclub.com/slack)
[x] I have self-reviewed my project proposal.
